server/server_test.go

```diff
@@ -0,0 +1,239 @@
+package server
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"pyrevittelemetryserver/cli"
+	"pyrevittelemetryserver/persistence"
+	"strings"
+	"testing"
+)
+
+type mockDB struct{ persistence.Connection }
+
+func setupTestRouter() http.Handler {
+	opts := &cli.Options{
+		ScriptsTable: "scripts",
+		EventsTable:  "events",
+	}
+	db := &mockDB{}
+	logger := &cli.Logger{}
+	return NewRouter(opts, db, logger)
+}
+
+func (m *mockDB) WriteScriptTelemetryV1(logrec *persistence.ScriptTelemetryRecordV1, logger *cli.Logger) (*persistence.Result, error) {
+	return &persistence.Result{ResultCode: 0, Message: "ok"}, nil
+}
+func (m *mockDB) WriteScriptTelemetryV2(logrec *persistence.ScriptTelemetryRecordV2, logger *cli.Logger) (*persistence.Result, error) {
+	return &persistence.Result{ResultCode: 0, Message: "ok"}, nil
+}
+func (m *mockDB) WriteEventTelemetryV2(logrec *persistence.EventTelemetryRecordV2, logger *cli.Logger) (*persistence.Result, error) {
+	return &persistence.Result{ResultCode: 0, Message: "ok"}, nil
+}
+func (m *mockDB) ReadScriptTelemetryV1(limit int, offset int, logger *cli.Logger) ([]persistence.ScriptTelemetryRecordV1, error) {
+	return []persistence.ScriptTelemetryRecordV1{}, nil
+}
+func (m *mockDB) ReadScriptTelemetryV2(limit int, offset int, logger *cli.Logger) ([]persistence.ScriptTelemetryRecordV2, error) {
+	return []persistence.ScriptTelemetryRecordV2{}, nil
+}
+func (m *mockDB) ReadEventTelemetryV2(limit int, offset int, logger *cli.Logger) ([]persistence.EventTelemetryRecordV2, error) {
+	return []persistence.EventTelemetryRecordV2{}, nil
+}
+func (m *mockDB) GetType() persistence.DBBackend {
+	return "mock"
+}
+func (m *mockDB) GetStatus(logger *cli.Logger) persistence.ConnectionStatus {
+	return persistence.ConnectionStatus{
+		Status:  "pass",
+		Version: "test",
+		Output:  "mock",
+	}
+}
+
+// -------------------------
+// Status Endpoint
+// -------------------------
+
+func TestStatusEndpoint(t *testing.T) {
+	router := setupTestRouter()
+
+	t.Run("GET /api/v1/status returns 200", func(t *testing.T) {
+		req := httptest.NewRequest("GET", "/api/v1/status", nil)
+		w := httptest.NewRecorder()
+		router.ServeHTTP(w, req)
+
+		if w.Code != http.StatusOK {
+			t.Errorf("expected 200 OK, got %d", w.Code)
+		}
+		if w.Body.Len() == 0 {
+			t.Errorf("expected non-empty body")
+		}
+	})
+}
```

Comment:
I know it is overkill for this simple endpoint, but there's also the /api/v2/status endpoint. Since it is identical, a for loop of "v1" and "v2" should cover everything.